sequencing/ATACSeq/jobSubmission/6_batchRunGenotypeCheck.sh

```diff
@@ -0,0 +1,157 @@
+mkdir -p $LOG_DIR
+mv ATACGenoConcorS6-${SLURM_ARRAY_JOB_ID}_${SLURM_ARRAY_TASK_ID}* $LOG_DIR
+
+## ================ ##
+##    VARIABLES     ##
+## ================ ##
+
+if [[ $2 == '' ]] 
+then
+  echo "No step specified. All steps in the script will be run. "
+fi
+
+##check array specified and exit if not
+if [[ ${SLURM_ARRAY_TASK_ID} == '' ]]
+then 
+    { echo "Job does not appear to be an array. Please specify --array on the command line." ; exit 1; }
+fi
+
+## check step method matches required options and exit if not
+if [[ ! $2 =~ "COMPARE" ]] && [[ ! $2 =~ "GENCHECK" ]] && [[ ! $2 =~ "SWITCH" ]] &&[[ ! $2 == '' ]];
+then 
+    { echo "Unknown step specified. Please use COMPARE, GENCHECK, SWITCH or some combination of this as a single string (i.e. SPLIT,CHECK)" ; exit 1; }            
+fi
+
+## ============ ##
+##    STEPS     ##
+## ============ ##
+
+## option COMPARE: genotype of samples are verified by comparing them with existing genotype information   
+if [ $# = 1 ] || [[ $2 =~ 'COMPARE' ]]
+then
+  module purge
+  module load $PICARDVERS
+  module load GATK
+  module load $SAMTVERS
+  
+  echo " "
+  echo "|| Running STEP 6.1 of ATAC-seq pipeline: COMPARE. Samples will be compared to their matching genotype data.||"
+  echo " "
+  echo "Output directory will be: ${ALIGNED_DIR}/genotypeConcordance"
+  echo "Output directory will be: ${ALIGNED_DIR}/baseRecalibrate"
+  echo " "
+  
+  # process a line from IDMap file
+  IDS=($(head -n ${SLURM_ARRAY_TASK_ID} ${META_DIR}/matchedVCFIDs.txt | tail -1))
+  echo "Samples to verify with their corresponding VCF IDs are: ${IDS[@]}"
+  
+  sh "${SUB_SCRIPTS_DIR}/compareBamWithGenotypes.sh" ${IDS[@]}
+  
+fi
+
+## option GENCHECK: Sex check and Genotype check results are collated in Rmarkdown   
+if [ $# = 1 ] || [[ $2 =~ 'GENCHECK' ]]
+then
+  
+	module purge
+	module load ${RVERS}
+	module load Pandoc
+ 
+  echo " "
+  echo "|| Running STEP 6.2 of ATAC-seq pipeline: GENCHECK. Results from genotype and sex check will be collated in a Rmarkdown report.||"
+  echo " "
+  echo "Output directory will be: ${PEAK_DIR}/QCOutput"
+  echo " "
+ 
+	Rscript -e "rmarkdown::render(paste0(commandArgs(trailingOnly=TRUE)[1], '/collateSampleChecks.Rmd'), output_file=paste0(commandArgs(trailingOnly=TRUE)[2], '/QCOutput/stage2SummaryStats.html'))" "${RSCRIPTS_DIR}" "$PEAK_DIR" "${CONFIGR}"
+  
+fi
+
+## option SWITCH: Samples with non-matching genotype are suggested to be switched
+if [ $# = 1 ] || [[ $2 =~ 'SWITCH' ]]
+then
+  module purge
+  module load $PICARDVERS
+  module load GATK
+  module load $SAMTVERS
+  
+  echo " "
+  echo "|| Running STEP 6.3 of ATAC-seq pipeline: SWITCH. Samples with potential genotype contamination will be selected for potential switches.||"
+  echo " "
+  echo "Output directory will be: ${ALIGNED_DIR}/genotypeConcordance"
+  echo "Output directory will be: ${ALIGNED_DIR}/baseRecalibrate"
+  echo " "
+  
+  cd ${ALIGNED_DIR}/genotypeConcordance/
```

Comment:
Does this directory necessarily exist? The only time I see the directory being created is in 
`compareBamWithGenotypes.sh` on line 94. However, that script is only being called if you enter `COMPARE` as the second argument to this script. If you don't have `COMPARE` but do have `SWITCH` in the second argument, this line will run (but the directory hasn't been created). 

I guess the directory might be created with line 42 in `searchBestGenoMatch.sh`. But this script won't have run yet (not until line 152 of this script). 

If this `cd` command fails, it looks like the only ramifications will be that lines 146 and 147 will likely fail (as they can't find these files).

Reply:
Then again, if you were to create this directory here, then the `awk` commands below will still fail as these `*.selfSM` files won't be found regardless (as the directory would be empty).
Am I missing something here? It feels like these `awk` commands will fail if the `COMPARE` section is not ran.

Reply:
This step of the script will only make sense to run if you have already run the COMPARE step, as it is where the results are output. The SWITCH step checks those samples that performed poorly in the COMPARE step and aims to find better genotype matches. That is why it is assumed that the genotypeConcordance directory exists already.

Reply:
Is this in the README?